Respect `adjust_name_casing` setting more consistently

SylvanSign opened the pull request:
- attempts to fix:
   - https://github.com/godotengine/godot/issues/87652
   - https://github.com/godotengine/godot/issues/87653
- I'm new to this repo, so not sure whether these are the best ways to fix those issues

* *Bugsquad edit, fixes: https://github.com/godotengine/godot/issues/87652*
* *Bugsquad edit, fixes: https://github.com/godotengine/godot/issues/87653*

Mickeon commented on scene/main/node.cpp:
```diff
@@ -1315,7 +1315,7 @@ void Node::_generate_serial_child_name(const Node *p_child, StringName &name) co
 		// No name and a new name is needed, create one.
 
-		name = p_child->get_class();
+		name = adjust_name_casing(p_child->get_class());
```
Hold on here. This also adjusts the name when `force_readable_name` is true in Node's `add_child`.

It's worth keeping in mind if this is desired (in my opinion, yes).

Mickeon commented on editor/plugins/node_3d_editor_plugin.cpp:
```diff
@@ -7561,6 +7561,7 @@ void Node3DEditor::_add_sun_to_scene(bool p_already_added_environment) {
 	}
 	ERR_FAIL_NULL(base);
 	Node *new_sun = preview_sun->duplicate();
+	new_sun->set_name(adjust_name_casing(new_sun->get_name()));
```
Did you not mean to fix the same issue for the Add Environment to Scene button? (It's created on the method right below this one).

FantasmaGolicon replied:
I am also just a first contributor, but putting my insight since I also looked at the issue without being aware there was a PR up already (And thought about doing this same solution):

Would it be preferred to fix the name for the WorldEnvironment on the calling point instead of modifying the Node implementation directly? I see that it is critical code, and it may not be worth the larger impact for an Editor-only fix. Along with the potential perf considerations.

Mickeon replied:
My thoughts exactly. As [I commented above](https://github.com/godotengine/godot/pull/87659#discussion_r1468838556), while the above solution may be more encompassing it would also affect code outside of the editor (even at run-time), which may have some unintended consequences. In this case it may be better to be "safe" and implement the same solution as the Sun node.

Mickeon reviewed (dismissed):
It's all good now, [could you squash your commits](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase)?

AThousandShips commented:
Your commit seems not to be linked to your GitHub account. See: [Why are my commits linked to the wrong user?](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user) for more info.

SylvanSign committed:
More consistent use of adjust_name_casing

- adjust_name_casing after duplicating preview sun
- adjust_name_casing after duplicating preview environment
- adjust_name_casing when creating Root Node from beginner_node_shortcuts

Mickeon commented:
The current solution would work fine and the PR should be kept as is, probably. But perhaps, to avoid repeating oneself, it could be delegated to a separate function or macro.

dalexeev commented on scene/main/node.h:
```diff
@@ -757,6 +757,8 @@ Error Node::rpc_id(int p_peer_id, const StringName &p_method, VarArgs... p_args)
 #define ERR_READ_THREAD_GUARD_V(m_ret)
 #endif
 
+#define SET_NAME_TO_CASE_ADJUSTED_CLASS(nodePtr) nodePtr->set_name(adjust_name_casing(nodePtr->get_class()))
```
I would prefer a method instead of a macro:

```c++
_FORCE_INLINE_ void set_name_to_case_adjusted_class() { set_name(adjust_name_casing(get_class())); }
```

It might make sense to name the method `reset_name()`, although this is too vague a name and might confuse someone into thinking the name will reset to something like `@Node@123`.

AThousandShips replied:
In either case it should be:
```suggestion
#define SET_NAME_TO_CASE_ADJUSTED_CLASS(m_node_ptr) m_node_ptr->set_name(adjust_name_casing(m_node_ptr->get_class()))
```

KoBeWi commented:
Forcing `adjust_name_casing()` with pascal case breaks names:
![image](https://github.com/godotengine/godot/assets/2223172/a900c7d1-e3cd-49dd-897f-af8f8fbe9935)
See this old issue: #65803

KoBeWi commented:
At least for `adjust_name_casing()`, we could add an argument whether it should skip pascal case. It's default almost everywhere, so it would save some checks.
